app/controllers/tournaments_controller.rb

```diff
@@ -1,16 +1,37 @@
+require 'json'
+
 class TournamentsController < ApplicationController
 
+  include TournamentsHelper
   skip_before_action :verify_authenticity_token
 
   def index
   end
 
   def create
-    Tournament.create!(company_name: params[:company_name], start_date: params[:start_date], end_date: params[:end_date])
+    begin
+      new_tournament = Tournament.create!(company_name: params[:company_name], start_date: params[:start_date], end_date: params[:end_date])
+      results = params[:results]
+      results.each do |result|
+        new_user = User.create!(first_name: result[:firstName], last_name: result[:lastName], email: result[:email])
+        temp = Result.create!(user_id: new_user.id, tournament_id: new_tournament.id , answers: result[:answers])
+      end
+      render json: { success: "New Tournament Created" }, status: :ok 
+    rescue 
+      render json: { error: "Internal Server Error" }, status: :internal_server_error
```

Comment:
Chances are that if we get to here, then something with the creation of either `Tournament`, `User` and `Result` failed. Since you're creating all three in this endpoint, it might be a good idea to let the caller know what failed (and why)

Luckily, Rails' ActiveRecord saves errors of certain models inside an `errors` property. Because you use a bang(!) method, i.e: `create!`, then if that operation failed, an "exception" is thrown (in Rails this is called an `error`). This error object contains the errors of why the creation failed.

Example:
```
rescue ActiveRecord::RecordInvalid => invalid
  render json: {error: "Failed to create a new tournament statistic, with error: #{invalid.record.errors}"}, status: :internal_server_error
```

If you were to use a standard `create` method, then the errors would be saved inside the model itself. IIRC it would be like this:
```
x = X.create(.....)
x.errors
```

Might be a good idea to fiddle around with this and see if it does indeed give the relevant information (i.e, try to fail the flow and see what you get back in the response)